apps/courses/factories.py

```diff
@@ -0,0 +1,96 @@
+"""
+Courses factories
+"""
+
+from random import randrange
+
+import factory
+
+from django.conf import settings
+from django.template.defaultfilters import slugify
+from django.utils import translation
+
+from apps.organizations.factories import OrganizationFactory
+
+from .models import Course, CourseSubject, CourseOrganizationRelation
+
+
+class CourseSubjectFactory(factory.django.DjangoModelFactory):
+    """
+        Django-parler compatible CourseSubject factory.
+        Will generate translation for current or default language, or can be called as follow:
+        `CourseSubjectFactory(translations__languages=['fr', 'en'])`
+    """
+
+    class Meta:
+        model = CourseSubject
+
+    @factory.post_generation
+    # pylint: disable=unused-argument, attribute-defined-outside-init
+    def translations(self, create, extracted, **kwargs):
+        """Create `django-parler` translation objects."""
+        # If no specific language specified, we use active one, then default one
+        current_language = [translation.get_language() or settings.LANGUAGE_CODE]
+        languages = kwargs.pop("languages", current_language)
```

Comment:
I think you should just take your languages from the languages keyword...
```python
languages = extracted or translation.get_language() or settings.LANGUAGE_CODE
```
So you can just write:
```python
CourseSubjectFactory(languages=['fr', 'en', 'de'])
```

Reply:
Sorry actually it should be: 
```python
languages = extracted or [translation.get_language() or settings.LANGUAGE_CODE]
```

Reply:
so be it. I was using `translations` because it is an actual field, but it may be a bad reason,
and this form is shorter

Reply:
Oh yeah I didn't realize that it was an existing field. 

My personal best practice for these post_generation fields is to call them `with_xxx` so I'm sure I have no collision with the object's fields.

Like this you can at the same time avoid using the name of an existing field and still avoid calling "field__action" when instanciating your factories.

Plus I find it makes a lot of sense when to write:
```python
CourseFactory(with_organisations=[...], with_languages=['fr', 'en'])
```